src/base/views/mouse-tool.ts

```diff
@@ -197,3 +200,24 @@ export class MouseListener {
         return vnode;
     }
 }
+
+@injectable()
+export class MousePositionTracker extends MouseListener {
+
+    protected lastPosition: Point | undefined;
+
+    mouseMove(target: SModelElement, event: MouseEvent): (Action | Promise<Action>)[] {
+        const viewport = findParentByFeature(target, isViewport);
```

Comment:
I think this should be generalized using `localToParent`, maybe we need another utility function along the lines of `getAbsoluteBounds` and `getAbsoluteClientBounds` from `features/bounds/model.ts`?